packages/langium/src/grammar/langium-grammar-validator.ts

```diff
@@ -291,6 +291,17 @@ export class LangiumGrammarValidator {
                         code: isInfers ? IssueCodes.SuperfluousInfer : IssueCodes.MissingInfer,
                         data: keywordNode && toDocumentSegment(keywordNode)
                     });
+                } else if(actionType && types.has(typeName) && isInfers) {
+                    // error: action infers type that is already defined
+                    if(action.$cstNode) {
+                        accept('error', `${typeName} is a declared type and cannot be redefined.`, {
+                            node: action,
+                            property: 'type',
+                            code: IssueCodes.SuperfluousInfer,
+                            // use the same data, stripping the 'infer' keyword out
+                            data: action.$cstNode.text.replace(/\s*infer\s+/, '')
```

Comment:
A specific call for feedback here. I was having some trouble deciding how to go about removing the infers. Initially I thought I was going to dig down into the keyword `infer` node in the Cst, and remove it entirely. However this ended up not working as expected. This approach feels cleaner than what I was doing previously, but I'm open to hear any alternatives that can more directly apply for removing the `infer` keyword.